### PR TITLE
Fixed farm+pipeline composition bug

### PR DESCRIPTION
**Pull request by mdolz**
Fixed farm+pipeline composition to support nested pipelines with more than 2 stages.

### CODE REVIEW

**Review by jdgarciauc3m — CHANGES_REQUESTED** [review]
Please, add a comment in the pull-request comments field explaining why decay_t is needed.

**Comment by mdolz** [comment]
std::decay_t removes the qualifiers in the first argument of both overloaded pipeline functions. If this is not done, both functors and execution policies return false in the "is_supported" trait and the compiler makes wrong decisions on what pipeline function should be choosen. This is because the "is_supported" traits expects a non-qualified execution policy type.

**Comment by jdgarciauc3m** [comment]
Why is this needed for pipeline and not for the rest of patterns?
Please provide an specific example of problematic calling code.

**Comment by mdolz** [comment]
Because it is the only pattern with two different interfaces.

```
  grppi::pipeline(e, 
    [x=1,n]() mutable -> optional<double> { 
      if (x<=n) return x++;
      else return {}; 
    },
    grppi::farm(4,
      grppi::pipeline(
        [](double x) { return x*x; },
        [](double x) { return x*x; },
        [](double x) { return 1/x; }
      )
    ),
    [](double x) { cout << x << endl; }
  );

```

File: include/pipeline.h [hunk 1]
```diff
@@ -47,13 +47,13 @@ namespace grppi {
...
 template <typename Execution, typename Generator, typename ... Transformers,
-          requires_execution_supported<Execution> = 0>
+          requires_execution_supported<std::decay_t<Execution>> = 0>
 void pipeline(
     const Execution & ex, 
     Generator && generate_op, 
     Transformers && ... transform_ops) 
 {
-  static_assert(supports_pipeline<Execution>(),
+  static_assert(supports_pipeline<std::decay_t<Execution>>(),
```
Comment (jdgarciauc3m) [inline include/pipeline.h:56]: Is this static_assert useful given the requires clause?

**Review by jdgarciauc3m — CHANGES_REQUESTED** [review]

File: include/pipeline.h (hunk 1 above)
Comment (mdolz) [reply include/pipeline.h:56]: The requires clause in the template parameters only checks if the Execution is an actual execution policy object, while the static_assert checks if the execution policy implements the pipeline pattern.

Comment (jdgarciauc3m) [reply include/pipeline.h:56]: Agreed! Thanks!

**Review by jdgarciauc3m — APPROVED** [review]
Thanks!

**Review by Drioastorga — APPROVED** [review]
Ok.

**Comment by jdgarciauc3m** [comment]
This solves issue #313

**Pull request merged**